.github/workflows/tests.yml

```diff
@@ -66,7 +66,7 @@ jobs:
           key: ${{ runner.os }}-cache
       # --- Code build ---
       - name: "Build dev tools"
-        run: inv -r faasmcli/faasmcli dev.tools --build Release
+        run: inv -r faasmcli/faasmcli dev.tools --clean --build Release
```

Comment:
This was the reason why the new tag wasn't being properly fetched from Git. It's still not clear to me what exactly is cached or not during this CI builds. The most I can say is that, empirically, adding this flag forces the fetch (otherwise only `CMake`' s populate step was triggered).

Reply:
AFACIT the issue is that CMake isn't detecting that the tag has changed on the external Faabric dependency, therefore we need to rerun the CMake configuration. Have you tried adding the following?

```
- name: "Rerun cmake config"
  run: inv -r faasmcli/faasmcli dev.cmake --build Release
```

The containers are structured so that we do a CMake build in the base container, then _rebuild_ it in the tests (rather than building from scratch). This is to avoid constantly downloading and rebuilding the dependencies.

The change here means that we completely nuke the existing CMake build, and build everything from scratch. So while it does work in picking up the new tag, it might add a couple of minutes to the build.

Reply:
`inv -r faasmcli/faasmcli dev.tools --build Release` already calls `cmake`under the hood, so adding those two lines won't make a difference.

I see your point, maybe we can discuss this in a bit today (in the meantime I'll do some reading on cmake).

Reply:
Ah yes sorry just seen this, in that case we should probably work out why it's not getting picked up...